### ISSUE TITLE
Unable to compile for electron@5.0.0-beta.1

### ISSUE THREAD

**Issue by vladimiry** [atom/node-keytar#151]
Build triggering command:
```
HOME=~/.electron-gyp node-gyp rebuild --target=5.0.0-beta.1 --arch=x64 --build-from-source --dist-url=https://atom.io/download/electron
```

Output:
```
gyp info it worked if it ends with ok
gyp info using node-gyp@3.8.0
gyp info using node@11.6.0 | linux | x64
gyp info spawn /usr/bin/python2
gyp info spawn args [ '/usr/lib/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/home/user/dev/project/node_modules/keytar/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/lib/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/user/.electron-gyp/.node-gyp/5.0.0-beta.1/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/home/user/.electron-gyp/.node-gyp/5.0.0-beta.1',
gyp info spawn args   '-Dnode_gyp_dir=/usr/lib/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/home/user/.electron-gyp/.node-gyp/5.0.0-beta.1/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/home/user/dev/project/node_modules/keytar',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/home/user/dev/project/node_modules/keytar/build'
  CXX(target) Release/obj.target/keytar/src/async.o
In file included from ../src/async.cc:4:
../node_modules/nan/nan.h: In function ‘void Nan::AsyncQueueWorker(Nan::AsyncWorker*)’:
../node_modules/nan/nan.h:1875:62: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
     , reinterpret_cast<uv_after_work_cb>(AsyncExecuteComplete)
                                                              ^
In file included from /home/user/.electron-gyp/.node-gyp/5.0.0-beta.1/include/node/node.h:67,
                 from ../node_modules/nan/nan.h:51,
                 from ../src/async.cc:4:
/home/user/.electron-gyp/.node-gyp/5.0.0-beta.1/include/node/v8.h: In instantiation of ‘void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = node::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)]’:
/home/user/.electron-gyp/.node-gyp/5.0.0-beta.1/include/node/node_object_wrap.h:85:78:   required from here
/home/user/.electron-gyp/.node-gyp/5.0.0-beta.1/include/node/v8.h:9555:16: warning: cast between incompatible function types from ‘v8::WeakCallbackInfo<node::ObjectWrap>::Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)’} to ‘Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<void>&)’} [-Wcast-function-type]
                reinterpret_cast<Callback>(callback), type);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/user/.electron-gyp/.node-gyp/5.0.0-beta.1/include/node/v8.h: In instantiation of ‘void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = Nan::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)]’:
../node_modules/nan/nan_object_wrap.h:66:61:   required from here
/home/user/.electron-gyp/.node-gyp/5.0.0-beta.1/include/node/v8.h:9555:16: warning: cast between incompatible function types from ‘v8::WeakCallbackInfo<Nan::ObjectWrap>::Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)’} to ‘Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<void>&)’} [-Wcast-function-type]
  CXX(target) Release/obj.target/keytar/src/main.o
In file included from ../src/main.cc:1:
../node_modules/nan/nan.h: In function ‘void Nan::AsyncQueueWorker(Nan::AsyncWorker*)’:
../node_modules/nan/nan.h:1875:62: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
     , reinterpret_cast<uv_after_work_cb>(AsyncExecuteComplete)
                                                              ^
../src/main.cc: At global scope:
../src/main.cc:45:15: error: variable or field ‘Init’ declared void
 void Init(v8::Handle<v8::Object> exports) {
               ^~~~~~
../src/main.cc:45:15: error: ‘Handle’ is not a member of ‘v8’
../src/main.cc:45:32: error: expected primary-expression before ‘>’ token
 void Init(v8::Handle<v8::Object> exports) {
                                ^
../src/main.cc:45:34: error: ‘exports’ was not declared in this scope
 void Init(v8::Handle<v8::Object> exports) {
                                  ^~~~~~~
In file included from ../node_modules/nan/nan.h:51,
                 from ../src/main.cc:1:
../src/main.cc:55:21: error: ‘Init’ was not declared in this scope
 NODE_MODULE(keytar, Init)
                     ^~~~
/home/user/.electron-gyp/.node-gyp/5.0.0-beta.1/include/node/node.h:506:36: note: in definition of macro ‘NODE_MODULE_X’
       (node::addon_register_func) (regfunc),                          \
                                    ^~~~~~~
../src/main.cc:55:1: note: in expansion of macro ‘NODE_MODULE’
 NODE_MODULE(keytar, Init)
 ^~~~~~~~~~~
../src/main.cc:55:21: note: suggested alternative:
 NODE_MODULE(keytar, Init)
                     ^~~~
/home/user/.electron-gyp/.node-gyp/5.0.0-beta.1/include/node/node.h:506:36: note: in definition of macro ‘NODE_MODULE_X’
       (node::addon_register_func) (regfunc),                          \
                                    ^~~~~~~
../src/main.cc:55:1: note: in expansion of macro ‘NODE_MODULE’
 NODE_MODULE(keytar, Init)
 ^~~~~~~~~~~
/home/user/.electron-gyp/.node-gyp/5.0.0-beta.1/include/node/node.h:217:18: note:   ‘node::Init’
 NODE_EXTERN void Init(int* argc,
                  ^~~~
In file included from /home/user/.electron-gyp/.node-gyp/5.0.0-beta.1/include/node/node.h:67,
                 from ../node_modules/nan/nan.h:51,
                 from ../src/main.cc:1:
/home/user/.electron-gyp/.node-gyp/5.0.0-beta.1/include/node/v8.h: In instantiation of ‘void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = node::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)]’:
/home/user/.electron-gyp/.node-gyp/5.0.0-beta.1/include/node/node_object_wrap.h:85:78:   required from here
/home/user/.electron-gyp/.node-gyp/5.0.0-beta.1/include/node/v8.h:9555:16: warning: cast between incompatible function types from ‘v8::WeakCallbackInfo<node::ObjectWrap>::Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)’} to ‘Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<void>&)’} [-Wcast-function-type]
                reinterpret_cast<Callback>(callback), type);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/user/.electron-gyp/.node-gyp/5.0.0-beta.1/include/node/v8.h: In instantiation of ‘void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = Nan::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)]’:
../node_modules/nan/nan_object_wrap.h:66:61:   required from here
/home/user/.electron-gyp/.node-gyp/5.0.0-beta.1/include/node/v8.h:9555:16: warning: cast between incompatible function types from ‘v8::WeakCallbackInfo<Nan::ObjectWrap>::Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)’} to ‘Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<void>&)’} [-Wcast-function-type]
In file included from ../src/main.cc:1:
../src/main.cc:38:12: warning: ‘Nan::NAN_METHOD_RETURN_TYPE {anonymous}::FindCredentials(Nan::NAN_METHOD_ARGS_TYPE)’ defined but not used [-Wunused-function]
 NAN_METHOD(FindCredentials) {
            ^~~~~~~~~~~~~~~
../node_modules/nan/nan.h:1326:33: note: in definition of macro ‘NAN_METHOD’
     Nan::NAN_METHOD_RETURN_TYPE name(Nan::NAN_METHOD_ARGS_TYPE info)
                                 ^~~~
../src/main.cc:31:12: warning: ‘Nan::NAN_METHOD_RETURN_TYPE {anonymous}::FindPassword(Nan::NAN_METHOD_ARGS_TYPE)’ defined but not used [-Wunused-function]
 NAN_METHOD(FindPassword) {
            ^~~~~~~~~~~~
../node_modules/nan/nan.h:1326:33: note: in definition of macro ‘NAN_METHOD’
     Nan::NAN_METHOD_RETURN_TYPE name(Nan::NAN_METHOD_ARGS_TYPE info)
                                 ^~~~
../src/main.cc:23:12: warning: ‘Nan::NAN_METHOD_RETURN_TYPE {anonymous}::DeletePassword(Nan::NAN_METHOD_ARGS_TYPE)’ defined but not used [-Wunused-function]
 NAN_METHOD(DeletePassword) {
            ^~~~~~~~~~~~~~
../node_modules/nan/nan.h:1326:33: note: in definition of macro ‘NAN_METHOD’
     Nan::NAN_METHOD_RETURN_TYPE name(Nan::NAN_METHOD_ARGS_TYPE info)
                                 ^~~~
../src/main.cc:15:12: warning: ‘Nan::NAN_METHOD_RETURN_TYPE {anonymous}::GetPassword(Nan::NAN_METHOD_ARGS_TYPE)’ defined but not used [-Wunused-function]
 NAN_METHOD(GetPassword) {
            ^~~~~~~~~~~
../node_modules/nan/nan.h:1326:33: note: in definition of macro ‘NAN_METHOD’
     Nan::NAN_METHOD_RETURN_TYPE name(Nan::NAN_METHOD_ARGS_TYPE info)
                                 ^~~~
../src/main.cc:6:12: warning: ‘Nan::NAN_METHOD_RETURN_TYPE {anonymous}::SetPassword(Nan::NAN_METHOD_ARGS_TYPE)’ defined but not used [-Wunused-function]
 NAN_METHOD(SetPassword) {
            ^~~~~~~~~~~
../node_modules/nan/nan.h:1326:33: note: in definition of macro ‘NAN_METHOD’
     Nan::NAN_METHOD_RETURN_TYPE name(Nan::NAN_METHOD_ARGS_TYPE info)
                                 ^~~~
make: *** [keytar.target.mk:113: Release/obj.target/keytar/src/main.o] Error 1
make: Leaving directory '/home/user/dev/project/node_modules/keytar/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:188:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:254:12)
gyp ERR! System Linux 4.20.3-1-MANJARO
gyp ERR! command "/usr/bin/node" "/usr/bin/node-gyp" "rebuild" "--target=5.0.0-beta.1" "--arch=x64" "--build-from-source" "--dist-url=https://atom.io/download/electron"
gyp ERR! cwd /home/user/dev/project/node_modules/keytar
gyp ERR! node -v v11.6.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
```

Compiling for electron@v4 works just fine with similar build triggering command:
```
HOME=~/.electron-gyp node-gyp rebuild --target=4.0.2 --arch=x64 --build-from-source --dist-url=https://atom.io/download/electron
```

**Comment by shiftkey** [comment]
I think I need a newer version of `nan` (#146 has the latest stuff) but I don't think that's enough yet as Electron 5 is aiming to ship with Node v12, which isn't quite stabilized yet - this isn't a priority for me in the short-term (v4 support is more urgent, but even that is blocked based on upstream issues).

**Comment by vladimiry** [comment]
> v4 support is more urgent, but even that is blocked based on upstream issues

Do you mean [this](https://github.com/atom/node-keytar/issues/133) issue? If yes, that's about using prebuilds while I'm interested in compiling from sources not using prebuilds.

**Comment by shiftkey** [comment]
`nan` currently only supports up to Node `11.4.0`, and I'm not sure how fixable these issues are on our side for a future version of Node.

**Comment by shiftkey** [comment]
@vladimiry I think the change merged in #152 should get this working, and has been released in [`4.3.1`](https://github.com/atom/node-keytar/releases/tag/v4.3.1) earlier today.

**Comment by vladimiry** [comment]
I confirm 4.3.1 release is compilable, thanks.

Couldn't try 4.3.1 in action though yet with electron@5.0.0-beta.1 as got blocked by https://github.com/electron/electron/issues/16631.
